File: 백준/Silver/1543. 문서 검색/문서 검색.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String n = br.readLine();
        String m  = br.readLine();
        int ans = 0;
        for(int i=0; i<=n.length()- m.length(); i++) {
            boolean flag = true;
            for(int j=0; j<m.length(); j++) {
                if(n.charAt(i+j) != m.charAt(j)) {
                    flag = false;
                    continue;
                }
            }
            if(flag) {
                i += (m.length()-1);
                ans++;
            }
        }
        System.out.println(ans);
    }
}

File: 백준/Silver/1543. 문서 검색/문서 검색.kt
val br = System.`in`.bufferedReader()
fun main() {
    val n = br.readLine()
    val m = br.readLine()
    var cnt = 0
    var idx = 0
    while (n.length >= idx + m.length) {
        if (n.substring(idx, idx + m.length) == m) {
            cnt++
            idx += (m.length - 1)
        }
        idx++
    }
    println(cnt)
}